components/prize_pool/commons/prize_pool.lua

```diff
@@ -0,0 +1,463 @@
+	self.widgetInjector = widgetInjector
+	return self
+end
+
+function PrizePool:setLpdbInjector(lpdbInjector)
+	-- TODO: Add type check once interface has been created
+	self.lpdbInjector = lpdbInjector
+	return self
+end
+
+function PrizePool:_hasUsdPrizePool()
+	return (Table.any(self.placements, function (_, placement)
+		return placement.hasUSD
+	end)) or (self.options.autoUSD and Table.any(self.prizes, function (_, prize)
+		return prize.enum == prizeTypes.LOCAL_CURRENCY
+	end))
+end
+
+--
+-- Static Functions
+--
+
+--- Remove all non-numeric characters from an input and changes it to a number.
+-- Most commonly used on money inputs, as they often contain , or .
+function PrizePool._parseInteger(input)
+	return tonumber((string.gsub(input, '[^%d.]', '')))
+end
+
+--- Checks if the input matches the format of a date
+function PrizePool._isValidDateFormat(date)
+	if String.isEmpty(date) then
+		return false
+	end
+	return string.match(date, '%d%d%d%d-%d%d-%d%d') and true or false
+end
+
+--- Checks that an Opponent Struct is valid and has a valid type
+function PrizePool._assertOpponentStructType(typeStruct)
+	if not typeStruct then
+		error('Please provide a type!')
+	elseif type(typeStruct) ~= 'table' or not typeStruct.type then
+		error('Could not parse type!')
+	elseif not Opponent.isType(typeStruct.type) then
+		error('Not a valid type!')
+	end
+end
+
+--- Fetches the LPDB object of a tournament
+function PrizePool._getTournamentInfo(pageName)
+	return mw.ext.LiquipediaDB.lpdb('tournament', {
+		conditions = '[[pagename::' .. pageName .. ']]',
+		limit = 1,
+	})[1]
+end
+
+--- Get the default date based on wiki-variables set in the Infobox League
+function PrizePool._getTournamentDate()
+	return Variables.varDefaultMulti('tournament_enddate', 'tournament_edate', 'edate', TODAY)
+end
+
+---
+function Placement:init(args, placeStart, opponentType)
+	self.args = self:_parseArgs(args)
+	self.date = self.args.date or PrizePool._getTournamentDate()
+	self.placeStart = self.args.placeStart
+	self.placeEnd = self.args.placeEnd
+	self.opponentType = opponentType
+	self.opponents = {}
+	self.prizes = {}
+
+	-- Parse prizes
+	self.prizes = self:_readPrizeRewards(self.args)
+
+	-- Parse opponents
+	self.opponents = self:_parseOpponents(self.args)
+
+	-- Implicit place range has been given (|place= is not set)
+	-- Use the last known place and set the place range based on the entered number of opponents
+	if not self.placeStart and not self.placeEnd then
+		self.placeStart = placeStart
+		self.placeEnd = placeStart + #self.opponents - 1
+	end
+
+	assert(self.placeStart and self.placeEnd, 'readPlacements: Invalid |place= provided.')
+	assert(#self.opponents > self.placeEnd - self.placeStart, 'readPlacements: Too many opponents')
+end
+
+function Placement:_parseArgs(args)
+	local parsedArgs = Table.deepCopy(args)
+	-- Explicit place range has been given
+	if args.place then
+		local places = Table.mapValues(mw.text.split(args.place, '-'), tonumber)
```

Comment:
use Array.map when working with arrays. makes it clearer what the inputs and outputs are.

if the output can be smaller than the input (due to f(x) returning nil) then use Array.collect to indicate that

Reply:
Array.map is not compatible with `tonumber`.